salt/hosts-minion/init.sls

```diff
@@ -0,0 +1,14 @@
+{%- for minion_id, addrlist in salt['mine.get']('roles:kube-master', 'network.ip_addrs', 'grain').items() %}
+{{ minion_id }}-host-entry:
+  host.present:
+{% if addrlist is string %}
+    - ip: {{ addrlist }}
+{% else %}
+    - ip: {{ addrlist|first }}
+{% endif %}
+    - names:
+      - api
+      - api.{{ pillar['internal_infra_domain'] }}
+      - {{ minion_id }}
+      - {{ minion_id }}.{{ pillar['internal_infra_domain'] }}
+{% endfor %}
```

Comment:
OK, but that will not be used - as soon as we add multiple masters we will be moving the `api` host to `127.0.0.1` and this will not be used. This works fine in a single master environment (which we are in right now), and will need an update regardless when we change that.

When we move to multi master, we will remove the 
```
      - api
      - api.{{ pillar['internal_infra_domain'] }}
```

lines and replace it with 

```
api-host-entry:
  host.present:
    - ip: 127.0.0.1
    - names:
      - api
      - api.{{ pillar['internal_infra_domain'] }}
```